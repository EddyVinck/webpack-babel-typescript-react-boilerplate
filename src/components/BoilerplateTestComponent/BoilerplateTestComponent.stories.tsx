import * as React from "react";
import { storiesOf } from "@storybook/react";
import BoilerplateTestComponent from "./BoilerplateTestComponent";

storiesOf("BoilerplateTestComponent", module).add("test render", () => (
  <BoilerplateTestComponent />
));
