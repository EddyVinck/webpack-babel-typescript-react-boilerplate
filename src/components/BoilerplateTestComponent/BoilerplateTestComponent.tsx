import * as React from "react";

import * as smallImage from "./assets/pexels-photo-270348.small.jpeg";
import * as largeImage from "./assets/pexels-photo-270348.jpeg";

class BoilerplateTestComponent extends React.Component {
  state = { elapsed: 0, ComponentTitle: null as React.ComponentType | null };

  private interval: number;

  componentDidMount() {
    this.interval = window.setInterval(() => {
      // requestAnimationFrame - verify that it is supported in Jest environment
      requestAnimationFrame(() => {
        this.setState({ elapsed: this.state.elapsed + 1 });
        // console.log("tick");
      });
    }, 1000);

    // async import - verify async loaded component loads correctly
    import(/* webpackChunkName: "asyncLoadedComponentTitle" */ "./AsyncLoadedComponentTitle").then(
      mod => {
        // console.log("### mod ###", mod);
        this.setState({ ComponentTitle: mod.default });
      },
    );
  }

  componentWillUnmount() {
    clearInterval(this.interval);
  }

  render() {
    /* tslint:disable:strict-boolean-expressions */
    return (
      <>
        {this.state.ComponentTitle && (
          <>
            <this.state.ComponentTitle />
            <br />
          </>
        )}
        Seconds: {this.state.elapsed}
        <br />
        <img src={smallImage} style={{ width: 100 }} />
        <br />
        <img src={largeImage} style={{ width: 100 }} />
      </>
    );
    /* tslint:enable:strict-boolean-expressions */
  }
}

export default BoilerplateTestComponent;
