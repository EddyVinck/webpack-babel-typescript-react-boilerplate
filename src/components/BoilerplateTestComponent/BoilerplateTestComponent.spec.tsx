import * as React from "react";
import * as renderer from "react-test-renderer";
import BoilerplateTestComponent from "./BoilerplateTestComponent";

it("renders", done => {
  const component = renderer.create(<BoilerplateTestComponent />);
  setTimeout(() => {
    expect(component.toJSON()).toMatchSnapshot();
  }, 1);
  setTimeout(() => {
    expect(component.toJSON()).toMatchSnapshot();
    done();
  }, 1200);
});
