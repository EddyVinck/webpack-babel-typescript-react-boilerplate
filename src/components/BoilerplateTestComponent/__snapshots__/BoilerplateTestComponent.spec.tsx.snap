// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
Array [
  "App Component",
  <br />,
  "Seconds: ",
  "0",
  <br />,
  <img
    src={
      Object {
        "default": "test-file-stub",
      }
    }
    style={
      Object {
        "width": 100,
      }
    }
  />,
  <br />,
  <img
    src={
      Object {
        "default": "test-file-stub",
      }
    }
    style={
      Object {
        "width": 100,
      }
    }
  />,
]
`;

exports[`renders 2`] = `
Array [
  "App Component",
  <br />,
  "Seconds: ",
  "1",
  <br />,
  <img
    src={
      Object {
        "default": "test-file-stub",
      }
    }
    style={
      Object {
        "width": 100,
      }
    }
  />,
  <br />,
  <img
    src={
      Object {
        "default": "test-file-stub",
      }
    }
    style={
      Object {
        "width": 100,
      }
    }
  />,
]
`;
